sherpa/models/tests/test_parameter.py

```diff
@@ -170,24 +170,51 @@ def setUp_composite():
     return p, p2
 
 
-@pytest.mark.parametrize("op", (abs, operator.neg))
+@pytest.mark.parametrize("op", [abs, operator.neg,
+                                # Test a selection of numpy ufuncs
+                                np.abs, np.sin, np.expm1])
 def test_unop(op):
     porig = setUp_p()
     p = op(porig)
     assert isinstance(p, UnaryOpParameter)
     assert p.val == op(p.val)
 
+def test_unop_string():
+    '''Check that the string output is sensible.
+
+    This implicitly also checks the logic to define a the format string for
+    printing the model representation.
+    '''
+    porig = setUp_p()
+    assert repr(-porig) == "<UnaryOpParameter '-model.name'>"
+    assert repr(np.cos(porig)) == "<UnaryOpParameter 'cos(model.name)'>"
+
 
 @pytest.mark.parametrize("op", [operator.add, operator.sub, operator.mul,
                                 operator.floordiv, operator.truediv, operator.mod,
-                                operator.pow])
+                                operator.pow,
+                                # Test a random selection of numpy ufuncs
+                                np.add, np.divide, np.true_divide,
+                                np.heaviside, np.greater, np.arctan2])
 def test_binop(op):
     p, p2 = setUp_composite()
     for comb in (op(p, p2.val), op(p.val, p2),
                  op(p, p2)):
         assert isinstance(comb, BinaryOpParameter)
         assert comb.val == op(p.val, p2.val)
 
+def test_binop_string():
+    '''Check that the string output is sensible.
+
+    This implicitly also checks the logic to define a the format string for
+    printing the model representation.
+    '''
+    p, p2 = setUp_composite()
+    comp = p + p2
+    assert repr(comp) == "<BinaryOpParameter '(model.p + model.p2)'>"
+    comp = np.logaddexp2(p, p2)
+    assert repr(comp) == "<BinaryOpParameter 'logaddexp2(model.p, model.p2)'>"
+
```

Comment:
I think it would be useful to have a check for the string output of something like `1 + np.e**p.gamma`, in part to convince me that the opstr changes is working as expected (i.e. it's not affecting this one). It would be nice to have the first commit be tests, then the second one changes + new-tests-that-are-now-possible (or removing an xfail). Then it makes it easier to have more confidence in what has/hasn't changed. But it depends on how many of these new tests actually work without the corresponding changes.

Reply:
I re-ordered the commits and added a new test (to be run before the code changes go in), i.e. it works in 4.14.

Reply:
Need to go back and fix up the first commit...

Reply:
OK, I think I separated it now such that the tests work commit-by-commit.